### PR TITLE
feat: add logs for upgrading candidate region and updating metadata

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Add logs for (region server) upgrading candidate region
Add logs for (procedure) updating metadata

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

## Refer to a related PR or issue link (optional)
#2700

### CODE REVIEW

**Review by killme2008 — APPROVED** [review]
LGTM.

**Review by MichaelScofield — APPROVED** [review]

File: src/mito2/src/worker/handle_catchup.rs [hunk 1]
```diff
@@ -44,6 +46,7 @@ impl<S: LogStore> RegionWorkerLoop<S> {
 
         // Utilizes the short circuit evaluation.
         let region = if !is_mutable_empty || region.manifest_manager.has_update().await? {
+            info!("Reopening the region: {region_id}");
```
Comment (evenyag) [inline src/mito2/src/worker/handle_catchup.rs:49]: Print `is_mutable_empty`.

File: src/mito2/src/worker/handle_catchup.rs [hunk 2]
```diff
@@ -67,6 +70,8 @@ impl<S: LogStore> RegionWorkerLoop<S> {
         };
 
         let flushed_entry_id = region.version_control.current().last_entry_id;
+        info!("Trying to replay memtable for region: {region_id}");
```
Comment (evenyag) [inline src/mito2/src/worker/handle_catchup.rs:73]: print flushed entry id?

File: src/mito2/src/worker/handle_catchup.rs [hunk 3]
```diff
@@ -75,6 +80,10 @@ impl<S: LogStore> RegionWorkerLoop<S> {
             &region.version_control,
         )
         .await?;
+        info!(
+            "Region: {region_id} catchup finished, elapsed: {:?}. last entry id: {last_entry_id}, expected: {:?}.",
+            request.entry_id, timer.elapsed()
```
Comment (evenyag) [inline src/mito2/src/worker/handle_catchup.rs:85]: Maybe you should print the timer first......

File: src/meta-srv/src/procedure/region_migration/update_metadata/upgrade_candidate_region.rs [hunk 4]
```diff
@@ -144,6 +144,11 @@ impl UpdateMetadata {
         } = datanode_table_value.region_info.clone();
         let table_route_value = ctx.get_table_route_value().await?;
 
+        let region_distribution = region_distribution(&region_routes);
+        info!(
+            "Trying to update region routes to {region_distribution:?} for table: {}",
+            region_id.table_id()
```
Comment (evenyag) [inline src/meta-srv/src/procedure/region_migration/update_metadata/upgrade_candidate_region.rs:150]: question: Is it recommended to mix two format styles?
```suggestion
            "Trying to update region routes to {:?} for table: {}",
            region_distribution, region_id.table_id()
```

**Pull request merged**